### PR TITLE
Core: Improve error for null/unknown schema types in table creation

### LINKED ISSUE #2539: error message for failed table creation due to null columns is misleading
### Apache Iceberg version

0.10.0 (latest release)

### Please describe the bug 🐞

I am assuming the catalog is on V2 since the null type doesn't get converted to unknown. I am not sure how to check, but it's in s3tables
```
# reprex-ish
import pyarrow as pa
# catalog = load_catalog(name="muck")
example_schema = pa.schema(
    [
        pa.field("int64", pa.int64()),
        pa.field("n1", pa.null()),
    ]
)
tbl = catalog.create_table("muck_ns.abc", schema=example_schema)
```

error log incorrectly blames "type string" : 

**BadRequestError: InvalidInputException: Invalid input:Cannot parse type string to primitive: unknown (through reference chain: org.apache.iceberg.rest.requests.CreateTableRequest["schema"])**

### Willingness to contribute

- [ ] I can contribute a fix for this bug independently
- [x] I would be willing to contribute a fix for this bug with guidance from the Iceberg community
- [ ] I cannot contribute a fix for this bug at this time

### PR DESCRIPTION
**Pull request by 010Soham**
What changed
- Add a client-side validation error when a PyArrow `null` type is used with format version < 3, including the field path and a hint to use a concrete type or format-version 3.
- Add a unit test to verify the clear error message when converting a PyArrow schema with a null field.

Why
- Prevents misleading REST errors like “Cannot parse type string… unknown” and points directly to the offending field.

Testing
- make lint
- uv run pytest tests/catalog/test_base.py -k rejects_null_type -v
- make test (timed out at ~42%)

Closes #2539

### CODE REVIEW

File: tests/catalog/test_base.py [hunk 1]
```diff
@@ -207,6 +207,16 @@ def test_convert_schema_if_needed(
...
 
 
+def test_convert_schema_if_needed_rejects_null_type(catalog: InMemoryCatalog) -> None:
+    schema = pa.schema([pa.field("n1", pa.null())])
+    with pytest.raises(ValueError) as exc_info:
+        catalog._convert_schema_if_needed(schema)
+    message = str(exc_info.value)
+    assert "Null type" in message
+    assert "n1" in message
+    assert "format-version=3" in message
```
Comment (kevinjqliu) [inline tests/catalog/test_base.py:217]: this is the full error message
```
'Null type (pa.null()) is not supported in Iceberg format version 2. Field: n1. Use a concrete type (string, int, boolean, etc.) or set format-version=3.'
```

File: pyiceberg/io/pyarrow.py [hunk 2]
```diff
@@ -1435,6 +1435,13 @@ def primitive(self, primitive: pa.DataType) -> PrimitiveType:
         elif pa.types.is_null(primitive):
             # PyArrow null type (pa.null()) is converted to Iceberg UnknownType
             # UnknownType can be promoted to any primitive type in V3+ tables per the Iceberg spec
+            if self._format_version < 3:
+                field_path = ".".join(self._field_names) if self._field_names else "<root>"
+                raise ValueError(
+                    "Null type (pa.null()) is not supported in Iceberg format version "
+                    f"{self._format_version}. Field: {field_path}. "
+                    "Use a concrete type (string, int, boolean, etc.) or set format-version=3."
```
Comment (geruh) [inline pyiceberg/io/pyarrow.py]: nit: wdyt about future proofing with a message like: `Requires v3+ or use a concrete type.`

**Review by Fokko — APPROVED** [review]
Thanks for adding this check, I think this will help a lot of users.

We can merge this after resolving @geruh's comment 👍

File: pyiceberg/io/pyarrow.py (hunk 2 above)
Comment (010Soham) [reply pyiceberg/io/pyarrow.py]: thanks @geruh 
Updated the error message to mention format-version 3+

**Comment by 010Soham** [comment]
@Fokko 
Have added the changes prescribed @geruh 
let me know if that sounds good

**Comment by kevinjqliu** [comment]
Thanks for the PR @010Soham and @geruh @Fokko for the review
